wagtail/admin/viewsets/model.py

```diff
@@ -474,7 +474,7 @@ def search_fields(self):
         If set to ``None`` and :attr:`search_backend_name` is set to use a Wagtail search backend,
         the ``search_fields`` attribute of the model will be used instead.
         """
-        return self.index_view_class.search_fields
+        return self.UNDEFINED
```

Comment:
Similar reasoning as c36b891e35ae177ca11b680001f79080ad2b233a. I think we ought to replace all `self.index_view_class.foo_attr` with `ViewSet.UNDEFINED`, but I think that's out-of-scope for this PR.